v2/pkg/operator/local_stored_collector.go

```diff
@@ -59,24 +63,41 @@ func (o *LocalStorageCollector) OperatorImageCollector(ctx context.Context) ([]v
 		imageIndexDir := strings.Replace(hld[len(hld)-1], ":", "/", -1)
 		cacheDir := strings.Join([]string{o.Opts.Global.WorkingDir, operatorImageExtractDir, imageIndexDir}, "/")
 		dir = strings.Join([]string{o.Opts.Global.WorkingDir, operatorImageDir, imageIndexDir}, "/")
-		if _, err := os.Stat(cacheDir); errors.Is(err, os.ErrNotExist) {
-			err := os.MkdirAll(dir, 0755)
-			if err != nil {
-				return []v1alpha3.CopyImageSchema{}, err
-			}
-			src := dockerProtocol + op.Catalog
-			dest := ociProtocolTrimmed + dir
-			err = o.Mirror.Run(ctx, src, dest, "copy", &o.Opts, *writer)
-			writer.Flush()
-			if err != nil {
-				o.Log.Error(errMsg, err)
+
+		// CLID-27 ensure we pick up oci:// (on disk) catalogs
+		if strings.Contains(op.Catalog, ociProtocol) {
+			// truncate the prefix to get the directory
+			ociDir := strings.Split(op.Catalog, ociProtocol)
```

Comment:
Suggestion: Here you can use imgSpec, err := image.ParseRef(img.Image) instead of HasPrefix.
imgSpec gives you the pathComponents as well as the transport.
Normally we performed unit tests to ensure it works for oci transport.

Reply:
Same comment as @sherine-k here